qa/workunits/cephadm/test_iscsi_setup.sh

```diff
@@ -0,0 +1,97 @@
+# The result from the first run would cause gateway creation to fail when the return value is used
+# later on. It was likely specific to my env, but it doesn't hurt to run it twice anyway. This
+# was the case whether I ran it through cephadm shell or directly on the host machine.
+FQDN=$(python3 -c 'import socket; print(socket.getfqdn())')
+ISCSI_POOL=$(sudo cephadm shell -- ceph orch ls iscsi --format json | jq -r '.[] | .spec | .pool')
+ISCSI_USER="adminadmin"
+ISCSI_PASSWORD="adminadminadmin"
+
+# gateway setup
+
+container_gwcli() {
+    sudo podman exec -it ${ISCSI_CONT_ID} gwcli "$@"
+}
+
+container_gwcli /iscsi-targets create iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw
+# I've seen this give a nonzero error code with an error message even when
+# creating the gateway successfully, so this command is allowed to fail
+# If it actually failed to make the gateway, some of the follow up commands will fail
+container_gwcli /iscsi-targets/iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw/gateways create ${FQDN} ${NODE_IP} || true
+container_gwcli /disks create pool=${ISCSI_POOL} image=disk_1 size=2G
+container_gwcli /disks create pool=${ISCSI_POOL} image=disk_2 size=2G
+container_gwcli /iscsi-targets/iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw/hosts create iqn.1994-05.com.redhat:client1
+container_gwcli /iscsi-targets/iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw/hosts/iqn.1994-05.com.redhat:client1 auth username=${ISCSI_USER}  password=${ISCSI_PASSWORD}
+container_gwcli /iscsi-targets/iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw/hosts/iqn.1994-05.com.redhat:client1 disk add ${ISCSI_POOL}/disk_1
+container_gwcli /iscsi-targets/iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw/hosts/iqn.1994-05.com.redhat:client1 disk add ${ISCSI_POOL}/disk_2
+
+# client setup
+
+sudo dnf install -y iscsi-initiator-utils device-mapper-multipath
+
+echo "GenerateName=no" | sudo tee -a /etc/iscsi/initiatorname.iscsi
+echo "InitiatorName=iqn.1994-05.com.redhat:client1" | sudo tee -a /etc/iscsi/initiatorname.iscsi
+
+echo "node.session.auth.authmethod = CHAP" | sudo tee -a /etc/iscsi/iscsid.conf
+echo "node.session.auth.username = ${ISCSI_USER}" | sudo tee -a /etc/iscsi/iscsid.conf
+echo "node.session.auth.password = ${ISCSI_PASSWORD}" | sudo tee -a /etc/iscsi/iscsid.conf
+
+sudo tee -a /etc/multipath.conf > /dev/null << EOF
+devices {
+  device {
+          vendor                 "LIO-ORG"
+          product                "TCMU device"
+          hardware_handler       "1 alua"
+          path_grouping_policy   "failover"
+          path_selector          "queue-length 0"
+          failback               60
+          path_checker           tur
+          prio                   alua
+          prio_args              exclusive_pref_bit
+          fast_io_fail_tmo       25
+          no_path_retry          queue
+  }
+}
+EOF
+sudo systemctl restart multipathd
+
+sudo iscsiadm -m discovery -t st -p ${NODE_IP}
+sudo iscsiadm -m node -T iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw -l
```

Comment:
This turned out to be an issue with the initiatorname.iscsi file. We modify it so that it doesn't generate a name and then we set a name ourselves, but it seems that modification is happening too late into the test when iscsi has already selected a randomized name, leaving you with
```
[root@smithi087 ~]# cat /etc/iscsi/initiatorname.iscsi 
InitiatorName=iqn.1994-05.com.redhat:891ea578b2bc
GenerateName=no
InitiatorName=iqn.1994-05.com.redhat:client1
```
The easiest solution is likely to just move modification of this file to earlier in the test.

Reply:
still the same issue, even when setting up the initiatorname file before doing the gateway or client setup
```
[root@smithi073 ~]# cat /etc/iscsi/initiatorname.iscsi 
InitiatorName=iqn.1994-05.com.redhat:ff9689c5c54
GenerateName=no
InitiatorName=iqn.1994-05.com.redhat:client1
```

Reply:
Is the iscsi initiator deamon started already before the test is run? Maybe that's a possible issue.

Reply:
Just pushed an alternative approach where we go back of writing the file later in the test, but we overwrite what was in there previously and there is an additional restart of the iscsid service

Reply:
That seems reasonable. I hope this appraoch works!